extensions/json/src/main/java/io/deephaven/json/BoolOptions.java

```diff
@@ -81,11 +79,11 @@ public interface Builder extends ValueOptionsSingleValueBase.Builder<Boolean, Bo
         Builder onMissing(boolean onMissing);
 
         default Builder onNull(Boolean onNull) {
-            return onNull((boolean) onNull);
+            return onNull == null ? this : onNull((boolean) onNull);
```

Comment:
This feels weird to me. If the user is trying to be explicit about our default behavior, why should we ignore it?